packages/web-client/app/components/card-pay/create-merchant-workflow/prepaid-card-choice/index.ts

```diff
@@ -0,0 +1,119 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+import WorkflowSession from '@cardstack/web-client/models/workflow/workflow-session';
+import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import MerchantCustomizationService, {
+  MerchantCustomization,
+} from '@cardstack/web-client/services/merchant-customization';
+import { isLayer2UserRejectionError } from '@cardstack/web-client/utils/is-user-rejection-error';
+import {
+  RegisterMerchantOptions,
+  TransactionHash,
+} from '@cardstack/web-client/utils/web3-strategies/types';
+
+import { task, TaskGenerator } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { reads } from 'macro-decorators';
+
+// http://ember-concurrency.com/docs/typescript
+// infer whether we should treat the return of a yield statement as a promise
+type Resolved<T> = T extends PromiseLike<infer R> ? R : T;
```

Comment:
This is also defined in app/components/card-pay/issue-prepaid-card-workflow/preview/index.ts. Let's extract it and define it in one place.

Reply:
👍🏻 I extracted it to `utils/resolved` 😀